test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -0,0 +1,586 @@
+     * and that no visible keys exist in the bounded range. 
+     */
+    void
+    validate_unsuccessful_search_near_call(scoped_cursor &normal_cursor, const std::string &lower_key, const std::string &upper_key)
+    {
+        int ret;
+        bool above_lower_key, below_upper_key;
+
+        /* Retrieve the key at the default cursor. */
+        const char *key_default;
+        testutil_check(normal_cursor->get_key(normal_cursor.get(), &key_default));
+        std::string key_default_str = key_default;
+
+        /* Check if the normal cursor's key is below the range or above the range bound. */
+        bool start_upper = lower_key.length() == 0 || std::lexicographical_compare(lower_key.begin(), lower_key.end(), key_default_str.begin(), key_default_str.end());
+
+        /* Here we validate that there are no keys in the bounded range that the range cursor could have returned. */
+        while (true) {
+            /* Traverse backwards or forwards depending on where the normal cursor is positioned. */
+            ret = start_upper ? normal_cursor->prev(normal_cursor.get()) : normal_cursor->next(normal_cursor.get());
+            if (ret == WT_NOTFOUND)
+                break;
+            testutil_assert(ret == 0);
+
+            testutil_check(normal_cursor->get_key(normal_cursor.get(), &key_default));
+            key_default_str = key_default;
+            /* Asserted that the traveresed key is not within the range bound. */
+            above_lower_key = lower_key.length() == 0 || std::lexicographical_compare(lower_key.begin(), lower_key.end(), key_default_str.begin(), key_default_str.end());
+            below_upper_key = upper_key.length() == 0 || std::lexicographical_compare(key_default_str.begin(), key_default_str.end(), upper_key.begin(), upper_key.end());
+            testutil_assert(!(above_lower_key && below_upper_key));
+            
+            /* Optimisation to early exit, if we have traversed past all possible records in the range bound. */
+            if ((!start_upper && !below_upper_key) || (start_upper && !above_lower_key))
+                break;
+        }
+    }
+
+    void
+    populate(test_harness::database &database, test_harness::timestamp_manager *,
+      test_harness::configuration *config, test_harness::workload_tracking *) override final
+    {
+        /*
+         * The populate phase only creates empty collections. The number of collections is defined
+         * in the configuration.
+         */
+        int64_t collection_count = config->get_int(COLLECTION_COUNT);
+
+        logger::log_msg(
+          LOG_INFO, "Populate: " + std::to_string(collection_count) + " creating collections.");
+
+        for (uint64_t i = 0; i < collection_count; ++i)
+            database.add_collection();
+
+        logger::log_msg(LOG_INFO, "Populate: finished.");
+    }
+
+    void
+    insert_operation(test_harness::thread_context *tc) override final
```

Comment:
Do we need a custom `inert_operation`? Can the default be used?

Reply:
unfortunately I think we do, because default `insert_operaton` only inserts with numbers, we would like all alphanumeric characters.